lib/rules/key-spacing.js

```diff
@@ -224,7 +238,7 @@ module.exports = function(context) {
 
                     return groups;
                 }, [[]]).forEach(function(group) {
-                    verifyAlignment(group);
+                    verifyAlignment(group, node);
```

Comment:
Since `node` is available at the very beginning, we don't need to go through the expense of assembling property groups unless we already know the object contains multiple lines. Passing `node` into `verifyAlignment` and implementing the exception there confuses the method's purpose. To avoid the duplicated code I mentioned in another comment and keep the two verification types single-purpose, I would recommend:

1. Rename `verifyAlignment` to `verifyGroupAlignment`.
1. Extract the existing contents of the `ObjectExpression` handler into a new `verifyAlignment` function.
1. Extract the existing contents of the `Property` handler into a new `verifySpacing` function.
1. Supply `verifySpacing` as the handler for `Property` nodes.
1. Within the `ObjectExpression` handler, if `node.loc.start.line === node.loc.end.line`, then `verifySpacing`, else `verifyAlignment`.